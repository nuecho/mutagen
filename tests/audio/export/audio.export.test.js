const sh = require("shelljs");
const { readCsv } = require("../audio.test.helpers.js");

const DEPENDENCIES_FILE_PATH = getResourcePath("audio/export/dependencies.csv");
const OUTPUT_PREFIX = "/tmp/mutagen/test/export";

mutagen(`--env gax audio import ${DEPENDENCIES_FILE_PATH}`);

test(`[mutagen audio export --help] should print export usage`, () => {
  const { code, output } = mutagen(`--env gax audio export --help`);

  expect(output).toMatchSnapshot("export usage");
  expect(code).toBe(0);
});

test(`[mutagen audio export] should print usage when output file is not specified`, () => {
  const { code, output } = mutagen(`--env gax audio export`);

  expect(output).toMatchSnapshot("export usage");
  expect(code).not.toBe(0);
});

test(`[mutagen audio export] should export messages into a csv file`, () => {
  const outputFile = `${OUTPUT_PREFIX}/csv-only/output.csv`;
  const mutagenResult = mutagen(`--env gax audio export ${outputFile}`);

  expect(mutagenResult.output).toMatchSnapshot("export done");
  expect(mutagenResult.code).toBe(0);

  const csvFile = exec(`cat ${outputFile} | egrep '(messageArId)|(export_test)'`);
  expect(csvFile.output).toMatchSnapshot("csv file");
});

test(`[mutagen audio export --personality-ids=x] should export messages into a csv file only for selected personalities`, () => {
  const outputFile = `${OUTPUT_PREFIX}/personalities/output.csv`;
  const mutagenResult = mutagen(`--env gax audio export --personality-ids=11 ${outputFile}`);

  expect(mutagenResult.output).toMatchSnapshot("export done");
  expect(mutagenResult.code).toBe(0);

  const csvFileContent = readCsv(outputFile, "export_test");
  expect(csvFileContent).toMatchSnapshot("csv file");
});

test(`[mutagen audio export --with-audios] should export audio files`, () => {
  const outputDirectory = `${OUTPUT_PREFIX}/with-audios`;
  const mutagenResult = mutagen(`--env gax audio export --with-audios ${outputDirectory}/output.csv`);

  expect(mutagenResult.output).toMatchSnapshot("export done");
  expect(mutagenResult.code).toBe(0);

  const downloadedFiles = exec(`find ${outputDirectory} -type f -name *export_test* | sort`);
  expect(downloadedFiles.output).toMatchSnapshot("downloaded files");
});
