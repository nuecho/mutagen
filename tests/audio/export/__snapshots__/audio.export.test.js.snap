// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[mutagen audio export --help] should print export usage: export usage 1`] = `
Object {
  "stderr": "",
  "stdout": "Usage: export [-?] [--with-audios] [--personality-ids=personality-id[,
              personality-id]...]... outputFile
Export audio files from ARM
      outputFile              Output audio description file.
      --personality-ids=personality-id[,personality-id]...
                              Comma separated list of which personalities'
                                audios to export (10,11,12,...).
      --with-audios           Download messages' audio files.
  -?, -h, --help              Show this help message.
",
}
`;

exports[`[mutagen audio export --personality-ids=x] should export messages into a csv file only for selected personalities: csv file 1`] = `
Object {
  "stderr": "",
  "stdout": "name,type,description,messageArId,tenantId,11
\\"export_test music name 1\\",MUSIC,\\"this is the description for music name 1\\",<removed-for-testing-purposes>,1,\\"11/export_test music name 1.wav\\"
\\"export_test message name 1\\",ANNOUNCEMENT,\\"this is the description for message name 1\\",<removed-for-testing-purposes>,1,\\"11/export_test message name 1.wav\\"
\\"export_test music name 2\\",MUSIC,\\"this is the description for music name 2\\",<removed-for-testing-purposes>,1,
\\"export_test message name 2\\",ANNOUNCEMENT,\\"this is the description for message name 2\\",<removed-for-testing-purposes>,1,
\\"export_test music name 3\\",MUSIC,\\"this is the description for music name 3\\",<removed-for-testing-purposes>,1,\\"11/export_test music name 3.wav\\"
",
}
`;

exports[`[mutagen audio export --personality-ids=x] should export messages into a csv file only for selected personalities: export done 1`] = `
Object {
  "stderr": "",
  "stdout": "",
}
`;

exports[`[mutagen audio export --with-audios] should export audio files: downloaded files 1`] = `
Object {
  "stderr": "",
  "stdout": "/tmp/mutagen/test/export/with-audios/10/export_test message name 1.wav
/tmp/mutagen/test/export/with-audios/10/export_test message name 2.wav
/tmp/mutagen/test/export/with-audios/10/export_test music name 3.wav
/tmp/mutagen/test/export/with-audios/11/export_test message name 1.wav
/tmp/mutagen/test/export/with-audios/11/export_test music name 1.wav
/tmp/mutagen/test/export/with-audios/11/export_test music name 3.wav
",
}
`;

exports[`[mutagen audio export --with-audios] should export audio files: export done 1`] = `
Object {
  "stderr": "",
  "stdout": "",
}
`;

exports[`[mutagen audio export] should export messages into a csv file: csv file 1`] = `
Object {
  "stderr": "",
  "stdout": "name,type,description,messageArId,tenantId,10,11
\\"export_test music name 1\\",MUSIC,\\"this is the description for music name 1\\",1000,1,,\\"11/export_test music name 1.wav\\"
\\"export_test message name 1\\",ANNOUNCEMENT,\\"this is the description for message name 1\\",1001,1,\\"10/export_test message name 1.wav\\",\\"11/export_test message name 1.wav\\"
\\"export_test music name 2\\",MUSIC,\\"this is the description for music name 2\\",1002,1,,
\\"export_test message name 2\\",ANNOUNCEMENT,\\"this is the description for message name 2\\",1003,1,\\"10/export_test message name 2.wav\\",
\\"export_test music name 3\\",MUSIC,\\"this is the description for music name 3\\",1004,1,\\"10/export_test music name 3.wav\\",\\"11/export_test music name 3.wav\\"
",
}
`;

exports[`[mutagen audio export] should export messages into a csv file: export done 1`] = `
Object {
  "stderr": "",
  "stdout": "",
}
`;

exports[`[mutagen audio export] should print usage when output file is not specified: export usage 1`] = `
Object {
  "stderr": "",
  "stdout": "Missing required parameter: outputFile
Usage: export [-?] [--with-audios] [--personality-ids=personality-id[,
              personality-id]...]... outputFile
Export audio files from ARM
      outputFile              Output audio description file.
      --personality-ids=personality-id[,personality-id]...
                              Comma separated list of which personalities'
                                audios to export (10,11,12,...).
      --with-audios           Download messages' audio files.
  -?, -h, --help              Show this help message.
",
}
`;
